app/screens/reschedule_draft/reschedule_draft.test.tsx

```diff
@@ -0,0 +1,197 @@
+                registerComponentListener: registerComponentListenerMock,
+            }),
+        },
+    };
+});
+
+jest.mock('@context/server', () => ({
+    useServerUrl: jest.fn(),
+}));
+
+const SERVER_URL = 'https://appv1.mattermost.com';
+
+describe('RescheduledDraft', () => {
+    let database: Database;
+
+    const mockDraft = {
+        scheduledAt: moment().add(1, 'day').valueOf(),
+        toApi: jest.fn().mockResolvedValue({
+            scheduled_at: moment().add(1, 'day').valueOf(),
+        }),
+    } as unknown as ScheduledPostModel;
+
+    const baseProps = {
+        componentId: Screens.RESCHEDULE_DRAFT,
+        closeButtonId: 'close-button-id',
+        currentUserTimezone: {
+            useAutomaticTimezone: true,
+            automaticTimezone: 'America/New_York',
+            manualTimezone: '',
+        },
+        draft: mockDraft,
+    };
+
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+        database = server.database;
+    });
+
+    beforeEach(() => {
+        jest.mocked(useServerUrl).mockReturnValue(SERVER_URL);
+    });
+
+    it('should render correctly', () => {
+        const {getByTestId} = renderWithEverything(
+            <RescheduledDraft {...baseProps}/>, {database},
+        );
+
+        expect(getByTestId('edit_post.screen')).toBeTruthy();
+    });
+
+    it('should have navigation component registered on initialization', () => {
+        renderWithEverything(<RescheduledDraft {...baseProps}/>, {database});
+
+        // Verify navigation listener was registered
+        expect(Navigation.events().registerComponentListener).toHaveBeenCalledWith(
+            expect.any(Object),
+            baseProps.componentId,
+        );
+
+        // Check that the component registered with proper navigation handler
+        const registerCall = jest.mocked(Navigation.events().registerComponentListener).mock.calls[0][0];
+        expect(registerCall).toBeDefined();
+        expect(registerCall.navigationButtonPressed).toBeDefined();
+    });
+
+    it('Should enable save button when data changes', async () => {
+        jest.mocked(updateScheduledPost).mockResolvedValue({scheduledPost: {} as ScheduledPost, error: undefined});
+
+        const {getByTestId} = renderWithEverything(
+            <RescheduledDraft {...baseProps}/>,
+            {database},
+        );
+
+        const dateTimeSelector = getByTestId('custom_date_time_picker'); // Ensure testID is set in the component
+        expect(dateTimeSelector).toBeTruthy();
+
+        const newDate = moment().add(2, 'days');
+        await act(async () => {
+            fireEvent(dateTimeSelector, 'handleChange', newDate);
+        });
+
+        expect(setButtons).toHaveBeenCalled();
+
+        // Use jest.mocked for proper type inference
+        const setButtonsMock = jest.mocked(setButtons);
+        const mockCalls = setButtonsMock.mock.calls;
+
+        const lastCallIndex = mockCalls.length - 1;
```

Comment:
You can clear the mocks (setButtonsMock.clearMocks) before performing the action that will trigger this.

If there are more than one call because of executing the action, could mean you are calling it more than needed.

Reply:
Sure